src/test/kotlin/com/terraformation/backend/tracking/db/PlantingSiteStoreTest.kt

```diff
@@ -1475,6 +1476,187 @@ internal class PlantingSiteStoreTest : DatabaseTest(), RunsAsUser {
+      store.ensurePermanentClustersExist(plantingSiteId)
+
+      val plots = monitoringPlotsDao.findAll()
+
+      assertEquals(8, plots.size, "Number of monitoring plots created")
+      assertEquals(
+          setOf(1, 2), plots.map { it.permanentCluster }.toSet(), "Permanent cluster numbers")
+      assertEquals(1, plots.minOf { it.name!!.toInt() }, "Smallest plot number")
+      assertEquals(8, plots.maxOf { it.name!!.toInt() }, "Largest plot number")
+    }
+
+    @Test
+    fun `only creates nonexistent permanent clusters`() {
+      val gridOrigin = point(0.0, 0.0)
+      val siteBoundary =
+          Turtle.makeMultiPolygon(gridOrigin) {
+            east(201)
+            north(201)
+            west(201)
+          }
+      val existingPlotBoundary =
+          Turtle.makePolygon(gridOrigin) {
+            east(25)
+            north(25)
+            west(25)
+          }
+
+      val plantingSiteId = insertPlantingSite(boundary = siteBoundary, gridOrigin = gridOrigin)
+      insertPlantingZone(boundary = siteBoundary, numPermanentClusters = 3)
+      insertPlantingSubzone(boundary = siteBoundary)
+      insertMonitoringPlot(
+          boundary = existingPlotBoundary, permanentCluster = 2, permanentClusterSubplot = 1)
+
+      store.ensurePermanentClustersExist(plantingSiteId)
+
+      val plots = monitoringPlotsDao.findAll()
+      val plotsPerCluster = plots.groupBy { it.permanentCluster }.mapValues { it.value.size }
+
+      assertEquals(9, plots.size, "Number of monitoring plots including existing one")
+      assertEquals(
+          mapOf(1 to 4, 2 to 1, 3 to 4), plotsPerCluster, "Number of plots in each cluster")
+    }
+
+    @Test
+    fun `can use temporary plot from previous observation as part of cluster`() {
+      val gridOrigin = point(0.0, 0.0)
+      val siteBoundary =
+          Turtle.makeMultiPolygon(gridOrigin) {
+            east(51)
+            north(51)
+            west(51)
+          }
+
+      // Temporary plot is in the southeast corner of the cluster.
+      val existingPlotBoundary =
+          Turtle.makePolygon(gridOrigin) {
+            moveStartingPoint { east(25) }
+            east(25)
+            north(25)
+            west(25)
+          }
+
+      val plantingSiteId = insertPlantingSite(boundary = siteBoundary, gridOrigin = gridOrigin)
+      insertPlantingZone(boundary = siteBoundary, numPermanentClusters = 1)
+      insertPlantingSubzone(boundary = siteBoundary)
+      val existingPlotId = insertMonitoringPlot(boundary = existingPlotBoundary)
+
+      store.ensurePermanentClustersExist(plantingSiteId)
+
+      val plots = monitoringPlotsDao.findAll()
+      val existingPlot = plots.first { it.id == existingPlotId }
+
+      assertEquals(4, plots.size, "Number of monitoring plots including existing one")
+      assertEquals(1, existingPlot.permanentCluster, "Permanent cluster of existing plot")
+      assertEquals(2, existingPlot.permanentClusterSubplot, "Subplot number of existing plot")
```

Comment:
2 because this is in the SE corner?

Reply:
Yep. Will add a comment to clarify this.

Reply:
Thanks for the change, looks good to me.